src/Tiny_httpd.ml

```diff
@@ -1088,17 +1088,21 @@ let run (self:t) : (unit,_) result =
     while self.running do
       (* limit concurrency *)
       Sem_.acquire 1 self.sem_max_connections;
-      let client_sock, _ = Unix.accept sock in
-      self.new_thread
-        (fun () ->
-           try
-             handle_client_ self client_sock;
-             Sem_.release 1 self.sem_max_connections;
-           with e ->
-             (try Unix.close client_sock with _ -> ());
-             Sem_.release 1 self.sem_max_connections;
-             raise e
-        );
+      try
+        let client_sock, _ = Unix.accept sock in
+        self.new_thread
+          (fun () ->
+            try
+              handle_client_ self client_sock;
+              Sem_.release 1 self.sem_max_connections;
+            with e ->
+              (try Unix.close client_sock with _ -> ());
+              Sem_.release 1 self.sem_max_connections;
+              raise e
+          );
+      with e ->
+        Printf.eprintf "accept raised an exception: %s"
+           (Printexc.to_string e)
```

Comment:
```suggestion
      match Unix.accept sock with
      | client_sock, _ ->
        self.new_thread
          (fun () ->
            try
              handle_client_ self client_sock;
              Sem_.release 1 self.sem_max_connections;
            with e ->
              (try Unix.close client_sock with _ -> ());
              Sem_.release 1 self.sem_max_connections;
              raise e
          );
      | exception e ->
        Printf.eprintf "accept raised an exception: %s"
           (Printexc.to_string e)
```

Reply:
this way only exceptions raised in `accept` will be captured.

Reply:
Thread.create may raise exception too (ok, the documentation does not mention it, but there is probably a maximum number of threads?) and exception in the son thread will never reach the father thread.
If you want to keep the server alive in a burst of request reaching the maximum number of thread temporarely, 
the best is to catch both exception from Unix.accept and Thread.create ?